src/main/java/org/primefaces/component/datatable/DataTable.java

```diff
@@ -1074,4 +1075,10 @@ public String getWidth() {
     public void setWidth(String width) {
         getStateHelper().put(InternalPropertyKeys.width, width);
     }
+
+    public boolean hasActiveFilter() {
```

Comment:
Introduce in UITable instead:
```java
    default boolean isFilteringCurrentlyActive() {
        return getFilterByAsMap().values().stream().anyMatch(FilterMeta::isActive);
    }
```

Reply:
used FilterMeta::isActive before but changed because it did not work if global filter has been specified. e.g. showcase sample always has an empty string as filterValue therefore isActive returns true.

Reply:
@Rapster So the question is should the UITable method change to be the one here?

Reply:
Sorry I'm really confused about all the modifications that has been made on DT for the last couple of months, and also quite surprised about issues users are facing... As of now, I'm just trying to give little help on code review, I mentionned UITable since TreeTable should have same functionnalities

Reply:
I guess I am asking that @robtha is saying your currently `isFilteringCurrentlyActive` in UITable is not complete enough and his above detects empty global filters etc.  I am asking should his method replace `isFilterCurrentlyActive` in UITable?

Reply:
Then, the problem is in FilterMeta#isActive then? But that's strange... Empty string should be set null by FilterFeature. For now, isFilteringCurrentlyActive should be introduced with @robtha impl, and it will most probably be reused for issue like #6913